cmplxs/cmplxs.go

```diff
@@ -0,0 +1,679 @@
+// entries have the maximum absolute value, the first such index is returned. If the slice
+// is empty, MaxAbsIdx will panic.
+func MaxAbsIdx(s []complex128) int {
+	if len(s) == 0 {
+		panic("cmplxs: zero slice length")
+	}
+	max := math.NaN()
+	var ind int
+	for i, v := range s {
+		if cmplx.IsNaN(v) {
+			continue
+		}
+		if a := cmplx.Abs(v); a > max || math.IsNaN(max) {
+			max = a
+			ind = i
+		}
+	}
+	return ind
+}
+
+// MinAbs returns the minimum absolute value in the input slice.
+// If the slice is empty, MinAbs will panic.
+func MinAbs(s []complex128) complex128 {
+	return s[MinAbsIdx(s)]
+}
+
+// MinAbsIdx returns the index of the minimum absolute value in the input slice. If several
+// entries have the minimum absolute value, the first such index is returned. If the slice
+// is empty, MinAbsIdx will panic.
+func MinAbsIdx(s []complex128) int {
+	if len(s) == 0 {
+		panic("cmplxs: zero slice length")
+	}
+	min := math.NaN()
+	var ind int
+	for i, v := range s {
+		if cmplx.IsNaN(v) {
+			continue
+		}
+		if a := cmplx.Abs(v); a < min || math.IsNaN(min) {
+			min = a
+			ind = i
+		}
+	}
+	return ind
+}
+
+// Mul performs element-wise multiplication between dst
+// and s and stores the value in dst. Panics if the
+// lengths of s and t are not equal.
+func Mul(dst, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	for i, val := range s {
+		dst[i] *= val
+	}
+}
+
+// MulTo performs element-wise multiplication between s
+// and t and stores the value in dst. Panics if the
+// lengths of s, t, and dst are not equal.
+func MulTo(dst, s, t []complex128) []complex128 {
+	if len(s) != len(t) || len(dst) != len(t) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	for i, val := range t {
+		dst[i] = val * s[i]
+	}
+	return dst
+}
+
+// NearestIdx returns the index of the element in s
+// whose value is nearest to v.  If several such
+// elements exist, the lowest index is returned.
+// NearestIdx panics if len(s) == 0.
+func NearestIdx(s []complex128, v complex128) int {
+	if len(s) == 0 {
+		panic("cmplxs: zero length slice")
+	}
+	switch {
+	case cmplx.IsNaN(v):
+		return 0
+	case cmplx.IsInf(v):
+		return MaxAbsIdx(s)
+	}
+	var ind int
+	dist := math.NaN()
+	for i, val := range s {
+		newDist := cmplx.Abs(v - val)
+		// A NaN distance will not be closer.
+		if math.IsNaN(newDist) {
+			continue
+		}
+		if newDist < dist || math.IsNaN(dist) {
+			dist = newDist
+			ind = i
+		}
+	}
+	return ind
+}
+
+// Norm returns the L norm of the slice S, defined as
+// (sum_{i=1}^N s[i]^L)^{1/L}
+// Special cases:
+// L = math.Inf(1) gives the maximum absolute value.
+// Does not correctly compute the zero norm (use Count).
+func Norm(s []complex128, L float64) float64 {
+	// Should this complain if L is not positive?
+	// Should this be done in log space for better numerical stability?
+	//	would be more cost
+	//	maybe only if L is high?
+	if len(s) == 0 {
+		return 0
+	}
+	if L == 2 {
```

Comment:
`switch`

Reply:
Done

Will also need to be done in floats.